src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AspNetCore6Plus/BrowserInjectingStreamWrapper.cs

```diff
@@ -120,41 +139,55 @@ public override async ValueTask DisposeAsync()
-                !_context.Response.ContentType.Contains("charset=", StringComparison.OrdinalIgnoreCase));
-
-            if (!_isHtmlResponse.Value)
+            try
             {
-                _agent.CurrentTransaction?.LogFinest($"Skipping RUM injection: Not an HTML response. ContentType is {_context.Response.ContentType}");
-                return false;
+                if (!forceReCheck && _isHtmlResponse != null)
+                    return _isHtmlResponse.Value;
+
+                // we need to check if the active request is still valid
+                // this can fail if we're in the middle of an error response
+                // or url rewrite in which case we can't intercept
+                if (_context?.Response == null)
+                    return false;
+
+                // Requirements for script injection:
+                // * text/html response
+                // * UTF-8 formatted (either explicitly or no charset defined)
+                _isHtmlResponse =
+                    _context.Response.ContentType != null && 
+                    _context.Response.ContentType.Contains("text/html", StringComparison.OrdinalIgnoreCase) &&
+                    (_context.Response.ContentType.Contains("utf-8", StringComparison.OrdinalIgnoreCase) ||
+                     !_context.Response.ContentType.Contains("charset=", StringComparison.OrdinalIgnoreCase));
+
+                if (!_isHtmlResponse.Value)
+                {
+                    _agent.CurrentTransaction?.LogFinest($"Skipping RUM injection: Not an HTML response. ContentType is {_context.Response.ContentType}");
+                    return false;
+                }
+
+                // Make sure we force dynamic content type since we're
+                // rewriting the content - static content will set the header explicitly
+                // and fail when it doesn't match if (_isHtmlResponse.Value)
+                if (!_isContentLengthSet && _context.Response.ContentLength != null)
+                {
+                    _context.Response.ContentLength = null;
+                    _isContentLengthSet = true;
+                }
             }
-
-            // Make sure we force dynamic content type since we're
-            // rewriting the content - static content will set the header explicitly
-            // and fail when it doesn't match if (_isHtmlResponse.Value)
-            if (!_isContentLengthSet && _context.Response.ContentLength != null)
+            catch (Exception e)
             {
-                _context.Response.Headers.ContentLength = null;
-                _isContentLengthSet = true;
+                LogExceptionAndDisable(e);
             }
 
-            return _isHtmlResponse.Value;
+            return _isHtmlResponse ?? false;
         }
 
+        private void LogExceptionAndDisable(Exception e)
+        {
+            _agent.Logger.Log(Level.Error,
```

Comment:
The wrappers allow for a small number of exceptions before shutting down. Did we want to emulate that pattern here or do we know that if one error occurs we should bail since it will not be a one-off?

Reply:
I didn't try to emulate that behavior; if an exception occurs once, it's almost certainly going to happen multiple times.

Reply:
I think that bailing is fine. Usually with our wrappers if there's an error, there are almost always enough errors to shut them down. I have not seen a case where we didn't see enough errors to disable the wrapper.

Reply:
Works for me!